src/bedrock/api.js

```diff
@@ -11,13 +11,75 @@ module.exports = class BedrockRealmAPI extends RealmAPI {
     return { host, port: Number(port) }
   }
 
-  async getRealmFromInvite (realmInviteCode) {
+  async getRealmFromInvite (realmInviteCode, skipAccept = false) {
     if (!realmInviteCode) throw new Error('Need to provide a realm invite code/link')
     const clean = realmInviteCode.replace(/https:\/\/realms.gg\//g, '')
     const data = await this.rest.get(`/worlds/v1/link/${clean}`)
+    if (!data.member && !skipAccept) await this.acceptRealmInviteFromCode(realmInviteCode) // If the player isn't a member, accept the invite
     return new Realm(this, data)
   }
 
+  async getRealmInvite (realmId) {
+    const data = await this.rest.get(`/links/v1?worldId=${realmId}`)
+    return {
+      inviteCode: data[0].linkId,
+      ownerXUID: data[0].profileUuid,
+      type: data[0].type,
+      createdOn: data[0].ts,
+      inviteLink: data[0].url,
+      deepLinkUrl: data[0].deepLinkUrl
+    }
+  }
+
+  async refreshRealmInvite (realmId) {
+    const data = await this.rest.post('/links/v1', {
+      body: {
+        type: 'INFINITE',
+        worldId: realmId
+      }
+    })
+    return {
+      inviteCode: data.linkId,
+      ownerXUID: data.profileUuid,
+      type: data.type,
+      createdOn: data.ts,
+      inviteLink: data.url,
+      deepLinkUrl: data.deepLinkUrl
+    }
+  }
+
+  async getPendingInviteCount () {
+    return await this.rest.get('/invites/count/pending')
+  }
+
+  async getPendingInvites () {
+    const data = await this.rest.get('/invites/pending')
+    return data.invites.map(invite => {
+      return {
+        invitationId: invite.invitationId,
+        worldName: invite.worldName,
+        worldDescription: invite.worldDescription,
+        worldOwnerName: invite.worldOwnerName,
+        worldOwnerXUID: invite.worldOwnerUuid,
+        createdOn: invite.date
+      }
+    })
+  }
+
+  async acceptRealmInvitation (invitationId) {
+    await this.rest.put(`/invites/accept/${invitationId}`)
+  }
+
+  async rejectRealmInvitation (invitationId) {
```

Comment:
What's the difference between invitation ID and code?

Reply:
A Realm can have a single universal code which can be used by any player to add that Realm to their list.

You can also invite players individually, the invitationId is the id of that invite which you use to accept/reject